Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -842,6 +873,91 @@ namespace Multiplayer
         return true;
     }
 
+    bool MultiplayerSystemComponent::HandleRequest(
+        IConnection* connection,
+        [[maybe_unused]] const IPacketHeader& packetHeader,
+        MultiplayerPackets::SyncComponentMismatch& packet)
+    {
+        // If this is the first packet containing component version data from this connection id, start a new vector for tracking the data
+        if (!m_connectedAppsComponentVersions.contains(connection->GetConnectionId()))
+        {
+            m_connectedAppsComponentVersions[connection->GetConnectionId()] = AZStd::vector<ComponentVersionMessageData>();
+        }
+
+        // Populate a list of the other app's multiplayer components.
+        // Once we have received all their components, we can begin comparing.
+        auto& theirMultiplayerComponents = m_connectedAppsComponentVersions[connection->GetConnectionId()];
+        for (const auto& theirComponentVersion : packet.GetComponentVersions())
+        {
+            theirMultiplayerComponents.push_back(theirComponentVersion);
+        }
+
+        if (theirMultiplayerComponents.size() != packet.GetTotalComponentCount())
+        {
+            // There's more component version data to receive from this connection
+            return true;
+        }
+
+        // We've received all the component data from this connection
+        // Iterate over each component and see what's been added, missing, or modified
+        for (const auto& theirComponent : theirMultiplayerComponents)
+        {
+            // Check for modified components
+            AZ::HashValue64 localComponentHash;
+            if (GetMultiplayerComponentRegistry()->FindComponentVersionHashByName(
+                    theirComponent.m_componentName, localComponentHash))
+            {
+                
+                if (theirComponent.m_componentVersionHash != localComponentHash)
+                {
+                    AZLOG_ERROR(
+                        "Multiplayer component mismatch! %s has a different version hash. Please make sure both client and server have "
+                        "matching multiplayer components.",
+                        theirComponent.m_componentName.GetCStr());
+                }
+            }
+            else
+            {
+                // Connected machine is using a multiplayer component we don't have
+                AZLOG_ERROR(
```

Comment:
I am wondering if this should be an ebus to report on? This way it could be easy to add it to the Editor for example, if there is some odd reason when this fails.

Reply:
Yes! Events coming in the next PR along with on-screen debug information, and unit tests.